### PR TITLE
Swap out docker linking with user-defined network and Docker DNS

### PR DESCRIPTION
**Pull request by colszowka**

### CODE REVIEW

**Comment by colszowka** [comment]
@hinnerk This is ready for review and merge. I went over the systemd dependencies, which went faster then expected - basically only adjusting the dependencies for nginx made sense, see https://github.com/experimental-platform/platform-configure/pull/127/commits/2a5cae2928d59e326bec7b2202d63fafa2645c11

Other than that on sop01 this works fine for me, and passes all [integration tests](https://github.com/protonet/soul-integration)

File: services/docker-protonet-network.service [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Docker protonet bridge network initialization
+After=docker.service
+
+[Service]
+ExecStart=-/usr/bin/env docker network create --driver bridge protonet
+ExecStartPost=/bin/bash -c "docker network inspect protonet | jq '.[0].IPAM.Config[0].Gateway' --raw-output | cut -d '/' -f1 > /etc/protonet/system/dockerhost"
+KillMode=none
+Type=oneshot
```
Comment (hinnerk) [inline services/docker-protonet-network.service]: This unit won't be enabled on boot as it has no `[[Install]]` section. Is that the way it's supposed to be? @colszowka

Comment (colszowka) [inline services/docker-protonet-network.service]: No, it isn't. I'll have to hunt down where I took this from, I took one of the existing oneshot commands as a template

**Comment by colszowka** [comment]
@hinnerk https://github.com/experimental-platform/platform-configure/pull/127/commits/6f9b81bbbc144948c77ae00bbc9be2ac807c8ddb

**Pull request merged**